main.cpp

```diff
@@ -395,29 +197,29 @@ int main()
     // FloatType object instanciation and method tests
     FloatType ft (3.0f);
     // --------
-    std::cout << "Initial value of ft: " << *(ft.value) << std::endl;
+    std::cout << "Initial value of ft: " << float(ft) << std::endl;
```

Comment:
FIXME this cast should be unnecessary now that you have User-Defined Conversion functions in place
- [ ] remove C-style cast.